### PR TITLE
Four Kitchens Release - Sprint 41

### PR DESCRIPTION
**Pull request by cienvaras**
# READY FOR REVIEW

## Summary

- SHS-5118: Implementation: Improve styling of the "Add to calendar" dropdown
- SHS-5180: Regression: Link field should generate URL to file, not media entity
- SHS-4998: Add hb-raised-cards--uniform-height as checkbox on collections
- SHS-5178: Implementation: Update Ocean more link color
- SHS-5179: Optimize mobile image sizes

## Need Review By (Date)
01/15

## Urgency
medium

### SHS-5118

1. Visit a page like https://dlcl.stanford.edu/events/french-speaking-worlds-rejane-senac (local copy: https://dlcl.suhumsci.loc/events/french-speaking-worlds-rejane-senac) 
2. Verify that the theming adjustments match those in the [Figma design.](https://www.figma.com/proto/hOtD5MStjFIavUrWapHKHN/UX%2FDesign-Tickets-Work?type=design&node-id=640-102&t=LhLJzFZTEJQDsIJn-0&scaling=min-zoom&page-id=635%3A20&hide-ui=1)

### SHS-5180

1. Create a flexible page
2. Add a postcard component
3. In the the postcard link field, add a link to an existing media file
4. Publish and save the new page
5. Navigate to the new page
6. Verify that the following roles can access the file:
    a. Anonymous
    b. Authenticated
    c. Contributor
    d. Author
    e. Site Manager

### SHS-4998

1. Create new Flexible Page
2. Click on the `Add Component` button
3. Add a Collection component
4. Verify that clicking the `Raised Cards` checkbox causes the `Uniform Height` checkbox to appear, already enabled.
5. Set the `Items Per Row` to 2
6. Disable the `Uniform Height` checkbox
8. Using the `Add Component` button in the Collection component, add 2 Postcard components
9. Add a paragraph or two to the first component and add only one line to the second component
10. Save the new page and navigate to it
11. Verify that the postcards are different heights, but still have the styling from the `hb-raised-cards` class
12. Edit the new page
13. Click on the `Edit` button on the Collection component
14. Enable the `Uniform Height` checkbox
15. Save the page and navigate to it
16. Verify that the postcards are now the same height and still have the styling from the `hb-raised-cards` class
17. Check existing content with Collection components. Content with `Raised Cards` enabled should have `Uniform Height` enabled as well. The following were used to test during development:
   -   West site
     1.  `Understanding Attitudes in the West` (Raised Cards enabled)
     2. `Student Profiles` (Raised Cards disabled)
   - Planning site
    1. `Department External Reviews` (Raised Cards disabled)
    2. `Events` (Raised Cards enabled)

### SHS-5178

1. Set up local with archaeology and history sites, switched to HS Colorful theme, and using the Ocean color scheme. 
2. Observe the link contrast on More links with arrows and ensure they pass the color contrast tests:
https://archaeology.suhumsci.loc/academics/field-experience
3. Ensure that links and more links in "wells" (beige background areas) pass contrast tests:
https://history.suhumsci.loc/people/caroline-winterer
4. Ensure buttons with a filled background pass contrast tests (including on hover):
https://archaeology.suhumsci.loc/events/distinguished-lecture-series/gathering-shadows-material-things
5. Ensure buttons with a white background pass contrast tests (including on hover): 
See "SUNet Login" in the footer area on any page
6. Ensure "reset" button passes contrast tests (including on hover):
https://history.suhumsci.loc/events/upcoming-events
7. Ensure Pager link color contrast passes contrast tests:
https://history.suhumsci.loc/recent-news
8. Ensure that date cards have proper contrast:
https://history.suhumsci.loc/events/upcoming-events

### SHS-5179

1. Visit a page like: https://southasia.suhumsci.loc/
2. Note that there is now a preset for the smallest width here where there was not before: 
https://southasia.suhumsci.loc/admin/config/media/image-styles/manage/postcard_vertical_linked_465px_
3.  Make sure that images with the right dimensions are being loaded when visiting a page with vertical linked postcards (e.g. Homepage).

Before: 
Screenshot from 2024-01-06 13-55-04

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

### CODE REVIEW

File: docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.deploy.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
```
Comment (pookmish) [inline docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.deploy.php]: What is this file? i've never heard of a deploy.php file nor do I understand how the function is intended to be used. I don't find any documentation for deploy hooks.

If this is just a simple update hook, move it to an install or post_update file

**Review by pookmish — CHANGES_REQUESTED** [review]

File: docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.deploy.php (hunk 1 above)
Comment (cienvaras) [reply docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.deploy.php]: @pookmish  Deploy hooks are part of drush and are executed after the configuration is imported. There's more information here:
- https://www.drush.org/12.x/deploycommand/
- https://www.hashbangcode.com/article/drupal-9-different-update-hooks-and-when-use-them

We consider is the best way for cases like this, when the db updates depend on configuration that's exported in the same PR. However, if you consider that's better to use an update hook, please let us know and we can change it.

Comment (joegl) [reply docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.deploy.php]: It looks like deploy hooks are only part of drush and not part of core. There [is an issue](https://www.drupal.org/project/drupal/issues/3206497) in the queue to add them to core, but there's not a lot of traction on it. It seems really useful but I'm a little concerned it's only drush supported.

Comment (pookmish) [reply docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.deploy.php]: I agree with Joe. To add to that, we would have to modify the deployment process to add the drush command. Let's move it to an update/post_update hook.

Comment (cienvaras) [reply docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.deploy.php]: @joegl @pookmish Done, PR ready for review again. Thanks!

File: docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.install [hunk 2]
```diff
@@ -71,3 +71,52 @@ function hs_paragraph_types_update_9403() {
     ])
     ->execute();
 }
+
+/**
+ * Creates field configuration found in the sync directory.
+ */
+function hs_paragraph_types_update_10000() {
+  $config_directories = Settings::get('config_sync_directory');
+  $file_storage = new FileStorage($config_directories);
```
Comment (pookmish) [inline docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.install]: There is a service for this: `config.storage`

**Review by pookmish — CHANGES_REQUESTED** [review]
Just need to swap for a service.

File: docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.install (hunk 2 above)
Comment (cienvaras) [reply docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.install]: @pookmish Fixed.

**Review by pookmish — APPROVED** [review]

**Pull request merged**